test/tools/base64_test.cpp

```diff
@@ -18,69 +18,42 @@
-  assert_true(i_val >= 0);
-
-  assert_true(((sz_size - 4) == ((size_t)i_val)) ||
-              (((size_t)i_val) == sz_size));
-
-  sz_size = Base64decode_len(pc_buffer);
-
-  assert_true(sz_size == sizeof(ac_output));
-
-  assert_true((size_t)Base64decode(ac_output.data(), pc_buffer) ==
-              /*strlen(ac_input)*/ sizeof(ac_input) - 1);
-
-  assert_true(strncmp(ac_input, ac_output.data(), sizeof(ac_input)) == 0);
-
-  delete (pc_buffer);
-  // pc_buffer = nullptr;
-
-  std::string encoded(cloudevents::base64::encode(std::string(ac_input)));
-
-  (void)memset(ac_output.data(), 0, sizeof(ac_output));
-  std::string decoded(cloudevents::base64::decode(encoded));
-
-  assert_true(std::string(ac_input) == decoded);
-}
-
-int main([[maybe_unused]] int argc, [[maybe_unused]] const char** argv) {
-  TestSuite* suite = create_test_suite();
-
-  add_test_with_context(suite, Base64, base64_encode_decode);
-
-  return run_test_suite(suite, create_text_reporter());
+TEST(Base64, base64_encode_decode) {
+    
+    static const char inputData[] =
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghi"
+        "jklmnopqrstuvwxyz!@#$%^&*()";
+    size_t inputDataLen = sizeof(inputData);
+    
+    static const char encodeCharArr[] = "QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVowMTIzNDU2Nzg5YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXohQCMkJV4mKigpAA==";
+    size_t encodeCharArrLen = sizeof(encodeCharArr);
+
+    static const char InvalidEncodeCharArr[] = "QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVowMTIzNDU2Nzg5YWJjZGdferVmZ2hpamtsbW5vcHFyc3R1dnd4eXohQCMkJV4mKigpAA==";
+    size_t invalidEncodeCharArrLen = sizeof(InvalidEncodeCharArr);
+
+    std::string encodeString{"QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVowMTIzNDU2Nzg5YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXohQCMkJV4mKigp"};
+    std::string invalidEncodeData{"QUJDREVGR0hJSktMTU5PUFFSU1RVVldYWVowMTIzNDU2Nzg5YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXohQCMkJV4mKigprs"};
+
+    std::string encoded(uprotocol::tools::base64::base64encode(std::string(inputData)));
+    ASSERT_EQ(encoded, encodeString);
+
+    std::string decoded(uprotocol::tools::base64::base64decode(encoded));
+    ASSERT_EQ(std::string(inputData), decoded);
+
+    std::string encodedCharStr(uprotocol::tools::base64::base64encode((uint8_t *)inputData,
+                                                           inputDataLen));
+    ASSERT_EQ(std::string(encodeCharArr), encodedCharStr);
+    
+    std::string decodedCharStr(uprotocol::tools::base64::base64decode((uint8_t *)encodeCharArr,
+                                                           encodeCharArrLen));
+    ASSERT_EQ(decodedCharStr.data(), std::string(inputData));
+
+    ASSERT_NE(uprotocol::tools::base64::base64decode(invalidEncodeData), decoded);
+    std::string InvaliddecodedCharStr(uprotocol::tools::base64::base64decode((uint8_t *)InvalidEncodeCharArr,
+                                                                      invalidEncodeCharArrLen));
+    ASSERT_NE(InvaliddecodedCharStr, decodedCharStr);
```

Comment:
Have separate test method for different test cases.

Reply:
Updated